go/vt/vtgate/tx_conn.go

```diff
@@ -126,8 +132,75 @@ func (txc *TxConn) RollbackIfNeeded(ctx context.Context, err error, session *Saf
 	}
 }
 
-// multiGo executes the action for all shardSessions in parallel and returns a consolildated error.
-func (txc *TxConn) multiGo(shardSessions []*vtgatepb.Session_ShardSession, action func(*vtgatepb.Session_ShardSession) error) error {
+// Resume resumes the specified 2PC transaction.
+func (txc *TxConn) Resume(ctx context.Context, dtid string) error {
+	splits := strings.Split(dtid, ":")
+	if len(splits) != 4 {
+		return vterrors.FromError(vtrpcpb.ErrorCode_BAD_INPUT, fmt.Errorf("invalid dtid: %s", dtid))
+	}
+	target := &querypb.Target{
+		Keyspace:   splits[0],
+		Shard:      splits[1],
+		TabletType: topodatapb.TabletType_MASTER,
+	}
+	txid, _ := strconv.ParseInt(splits[3], 10, 0)
```

Comment:
I would get the error there too and return the BAD_INPUT thing.

Maybe also isolate the dtid parsing into its own function? Maybe also the printing of a dtid? With unit tests?